### PR TITLE
[WIP] AWS event subscription(DynamoDB)

### PR DESCRIPTION
**Pull request by supritat**
### Description of the Change

* Included custom role in serverless.yml file
* Including policy list that is required for DynamoDB stream
* Jenkins file is updated

### Benefits

AWS event subscription using dynamoDb stream.

### CODE REVIEW

**Comment by SiniWilson** [comment]
Please change the base branch from v1.8.1  to events

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 1]
```diff
@@ -281,6 +285,45 @@ node() {
...
+def createDynamodbStream(tableName, region) {
+	def tableDetails = sh(
+    script: "aws dynamodb update-table --table-name ${tableName} --stream-specification StreamEnabled=true,StreamViewType=NEW_AND_OLD_IMAGES --region ${region} --output json",
+    returnStdout: true
+  ).trim()
+  def tableDetailsJson = parseJson(tableDetails)
+
+	return tableDetailsJson.TableDescription.LatestStreamArn
+}
+
```
[the remark this answers is not in the document]
Comment (supritat) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: Included in <https://github.com/tmobile/jazz/pull/510>

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 2]
```diff
@@ -583,6 +629,14 @@ def writeServerlessFile(config, current_environment) {
 	if (config['mainClass']) {
 		sh "sed -i -- 's/\${file(deployment-env.yml):mainClass}/${config['mainClass']}/g' serverless.yml"
 	}
+  if (config['event_source_dynamodb']) {
+    sh "sed -i -- 's/resourcesDisabled/resources/g' ./serverless.yml"
+    sh "sed -i -- 's|\${file(deployment-env.yml):event_source_dynamodb}|${config['event_source_dynamodb']}|g' serverless.yml"
+    sh "sed -i -- 's|\${file(deployment-env.yml):event_source_dynamodb}|${config['event_source_dynamodb']}|g' policyFile.yml"
+    sh "sed -i -e 's|\${file(deployment-env.yml):iamRoleARN}|customEventRole|g' serverless.yml"
```
[the remark this answers is not in the document]
Comment (supritat) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: Included in <https://github.com/tmobile/jazz/pull/510>

File: builds/serverless-config-pack/policy.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                - "dynamodb:GetRecords"
+                - "dynamodb:GetShardIterator"
+                - "dynamodb:ListStreams"
+              Resource: "${file(deployment-env.yml):event_source_dynamodb}"
+            - Effect: "Allow"
+              Action:
+                - "logs:CreateLogGroup"
+                - "logs:CreateLogStream"
+                - "logs:PutLogEvents"
+              Resource: "*"
```
[the remark this answers is not in the document]
Comment (supritat) [reply builds/serverless-config-pack/policy.yml]: Need to discuss

**Comment by supritat** [comment]
Creating new PR with mentioned comments.
Follow up PR: <https://github.com/tmobile/jazz/pull/510>

**Pull request closed without merging**